### DISCUSSION TITLE
`UnusedCall` opt-in rather than opt-out?

### DISCUSSION THREAD

**Discussion by a-frantz** [stjude-rust-labs/wdl#250] [Feature requests]
Asking here instead of in Issues because I'm not 100% sure this is the correct solution.

The issue: the `UnusedCall` diagnostic is often a false positive. There are valid/common use cases for calling a task and ignoring it's outputs.

Potential solution:
- do not emit `UnusedCall` by default
- new lint directive of the form `#@ include: UnusedCall` which can only be added to the preamble of files with a workflow
    - (is "include" the right keyword?)
    - For `workflows`, we would add this new directive to some of our files. i.e. there would be at least one user of this new syntax and it's us lol
- Once we have some sort of `~/.config/wdl` file, this can always be toggled on for a user

tagging @adthrasher and @peterhuene for thoughts on this?

**Comment by adthrasher** [comment]
> new lint directive of the form #@ include: UnusedCall which can only be added to the preamble of files with a workflow

If we went this route, would users know on which files to enable this? They wouldn't be getting the diagnostics, so they'd have to determine via some other method to add the directive.

**Reply by a-frantz** [reply]
Good point that this feature has essentially zero discoverability.... really the only way to enable this feature would be to read through sprocket documentation that doesn't currently exist.

But _maybe_ that's ok? If we think of this as some "advanced feature" isn't it ok for it to only be discovered via reading the docs?

I don't have any evidence here, but I would think this diagnostic is considered an annoying false positive more often than it's considered a useful warning.
